### PR TITLE
Fix running extensions with relative paths

### LINKED ISSUE #586: Cannot run extensions with relative paths
When using `phylum extension run` to execute an extension without installing it, it will print out an error when the extension path is relative:

```
$ extension run ./duplicates
❗ Error: Execution failed caused by: `/tmp/duplicates/main.ts`: importing from paths outside of the extension's directory is not allowed
```

### PR DESCRIPTION
**Pull request by cd-work**
This patch fixes an issue with `phylum extension run`, where relative
paths were not accepted as parameter.

To ensure that our module loader's restriction to only load from within
the extension's directory still works properly, all paths are not
canonicalized in the module loader. This means we do not need to check
for symlinks explicitly.

Closes #586.

### CODE REVIEW

**Review by andreaphylum — DISMISSED** [review]

**Comment by cd-work** [comment]
I've run this through a bunch of manual tests and things still seem to work fine:
 - Symlinks work when they're inside the extension directory
 - Imports work when they're inside the extension directory
 - Symlinks don't work when they point outside the extension directory
 - Imports don't work when they point outside the extension directory
 - Missing modules spit out an error (error improved, see below)

All the errors still seem to make sense, I haven't noticed any changes.
Errors for missing modules have been improved, not by much but I think that's the least we can do:

Before:

```
❗ Error: Execution failed caused by: No such file or directory (os error 2)
```

After:

```
❗ Error: Execution failed caused by: Invalid extension module: "/tmp/testing/tes.ts"
```

@andreaphylum you've messed with this before and decided to go against canonicalization, now the same problem popped up again. I think it's a good choice to just get it sorted out with canonicalization but I'd still like your review to ensure I didn't miss an issue you found in your previous explorations.

File: cli/tests/extensions/module_loader.rs [hunk 1]
```diff
@@ -56,23 +57,3 @@ fn module_with_non_allowed_url_fails_to_load() {
...
-    test_cli.install_extension(&fixtures_path().join("symlink")).success();
-
-    std::os::unix::fs::symlink(ext_path.join("symlink_me.ts"), ext_path.join("symlink.ts"))
-        .unwrap();
-
-    test_cli
-        .run(&["symlink"])
-        .failure()
-        .stderr(predicate::str::contains("importing from symlinks is not allowed"));
-}
```
Comment (andreaphylum) [inline cli/tests/extensions/module_loader.rs:78]: Why was this removed? If we want it removed, we should probably also remove the associated fixture.

**Review by andreaphylum — DISMISSED** [review]
Looks good to me. Having both paths canonicalized in `deno.rs` should be enough to allow us to determine the prefix, and I don't think we need to do that anywhere else.

If we want to be more confident, we could try and formalize some of the manual tests you mentioned into automated tests, but I think the current setup is sufficient to cover most regressions, too.

File: cli/src/config.rs [hunk 2]
```diff
@@ -247,6 +247,7 @@ mod tests {
         assert_eq!(config.auth_info.offline_access, Some(RefreshToken::new(ENV_TOKEN)));
     }
 
+    #[ignore]
```
[the remark this answers is not in the document]
Comment (cd-work) [reply cli/src/config.rs:250]: Leftover from testing, since it's broken. Will remove.

**Comment by cd-work** [comment]
> If we want to be more confident, we could try and formalize some of the manual tests you mentioned into automated tests, but I think the current setup is sufficient to cover most regressions, too.

I think a lot of the test are already covered by our existing extension module loader tests. The main goal of the manual tests was to ensure error output and formatting made sense.

**Review by andreaphylum — APPROVED** [review]
